docs: adds in information on how to test PRs in GitHub Codespaces (#2617)
trussworks/react-uswds: docs/codespace_pr_reviews.md

# Using Codesapces to review PRs without pulling a repo to your machine

> 

Reviewer: I'm assuming that the correct versions of packages etc will run within codespaces, so it's not necessary to have those updated locally. If that's correct, perhaps a short blurb explaining that would be helpful here.

Author: I added that context in, explaining that is *why* you have to wait a while when you first start the codespace.

1. go to the PR
2. click the `Code` button in the top right corner
3. click the `Codespaces` tab
4. `Create new codespace`

>    ![Create a new codespace screenshot](https://prod-files-secure.s3.us-west-2.amazonaws.com/bf5cb10c-8e7c-41db-97ca-ded0d1f001c9/49ab6248-8f67-4e79-9ee5-7dee09e03237/Untitled.png)

Reviewer 2: Odd, i'm not seeing the exact language from your screenshot 
<img width="385" alt="image" src="https://github.com/trussworks/react-uswds/assets/764090/54bd678c-77e5-46d0-b008-bbb41ca61de4">

Author: in my screenshot, I had had a codespace on the branch previously, i didn't realize that would make a difference. do you think it's confusing?

Reviewer 2: same, i created then deleted. Probably fine and followable and likely to change again anyway

Author: i see what you mean now, I've updated it to reflect "create a new codespace on" language to make it less confusing.

> 5. **Wait** for the codespace to finish setting up, you should see stuff happening in the `Terminal`. If you mess with it before it is done it might not work. If you mess this up, yo might see an error about your node version.

Reviewer 2: ```suggestion
5. **Wait** for the codespace to finish setting up, you should see stuff happening in the `Terminal`. If you mess with it before it is done it might not work. If you mess this up, you might see an error about your node version.
```

6. if the yarn install in that process was not successful, run `yarn install`. Wait until it is done before you do anything. takes ~5ish minutes

> 7. if it is successful run `yarn storybook`

Reviewer: Running it this first time, mine got stuck on 99% done:
<img width="698" alt="image" src="https://github.com/trussworks/react-uswds/assets/13249580/b9d66c78-3659-4c18-a7b4-c6d493354135">

Reviewer: re conversation on Slack- Yes, adding the screenshot of the port and pointing to the globe as the link would be helpful.

8. You will get a message saying your application is running on port 9009. click the `Open in Browser` button

>    ![application running message](https://github.com/trussworks/react-uswds/assets/59394696/9d6ac91d-5ea2-4bbc-a64d-685076444354)

Reviewer: Where does this button appear? In the terminal? or somewhere else?

# Using Codesapces to review PRs without pulling a repo to your machine

1. go to the PR
2. click the `Code` button in the top right corner
3. click the `Codespaces` tab
4. `Create a new codespace on ...`

   ![Create a new codespace screenshot](https://github.com/trussworks/react-uswds/assets/59394696/9ebfff8a-353c-4064-bf70-8712bbfbbeda)

5. **Wait** for the codespace to finish setting up, you should see stuff happening in the `Terminal`. This is the codespace installig the correct versions of packages etc will run in the environment, so it's not necessary to have those updated locally, or even have the repo on your machine locally at all. If you mess with it before it is done it might not work. If you mess this up, you might see an error about your node version.

   When it is complete, you should see a message like this in the terminal. If you don't see this, you should delete the codespace and start over.

   ![Github Codespace configuaration process](https://github.com/trussworks/react-uswds/assets/59394696/bdf6cd8d-22e2-4e6b-b778-65c83c51556a)

> 6. if the yarn install in that process was not successful, like in the image above, run `yarn install`. Wait until it is done before you do anything. takes ~5ish minutes

Reviewer: I first read this as meaning "was not successful, like in the image [showing unsuccess] above".

```suggestion
6. if the yarn install in that process was not successful (should look like the image above), run `yarn install`. Wait until it is done before you do anything. takes ~5ish minutes
```

7. if it is successful run `yarn storybook`. this might also take a couple of minutes to complete.
8. when the link the storybook will be hosted on is available, you will get a message saying your application is running on port 9009 in the bottom right corner. click the `Open in Browser` button

   ![application running message](https://github.com/trussworks/react-uswds/assets/59394696/9d6ac91d-5ea2-4bbc-a64d-685076444354)

>    If you don't see that message (it doesnt alawys happen) check the terminal for something like this:

Reviewer 3: ```suggestion
   If you don't see that message (it doesn't always happen) check the terminal for something like this:
```

   ![storybook started message in the terminal](https://user-images.githubusercontent.com/13249580/272653168-b9d66c78-3659-4c18-a7b4-c6d493354135.png)

>    If you see that, go to `Ports` tab in the top bar of that bottom section where your `Terminal` is. You can find a link under the column in the table labelled `Forwarded address`. Click that to open the sotrybook site being hosted on Codespaces.

Reviewer 3: ```suggestion
   If you see that, go to `Ports` tab in the top bar of that bottom section where your `Terminal` is. You can find a link under the column in the table labelled `Forwarded address`. Click that to open the Storybook site being hosted on Codespaces.
```

   ![Ports Tab table with forwarded address](https://github.com/trussworks/react-uswds/assets/59394696/a3075661-a1c2-476a-aeb9-974e6f9e036f)

9. It should open a new tab that will have a blank window until the storybook is completely done running its build. Might take a couple of minutes.

> 10. When the build completes, should show you the react uswds Storybook with the changes in the PR!

Reviewer 3: ```suggestion
10. When the build completes, should show you the ReactUSWDS Storybook with the changes in the PR!
```

Reviewer 3: This is at least how we refer to things in our main README: https://github.com/trussworks/react-uswds/blob/main/README.md

    ![storybook running in codespaces dev environment](https://github.com/trussworks/react-uswds/assets/59394696/cdfc67de-711f-43c2-928a-284158744f1f)

11. proceed as if you were running the PR on your own machine. hooray!

> 12. When your review is complete, you should delete the codespace. There are costs associated with storing codespaces. **You should therefore delete any codespaces you no longer need**.

Reviewer 3: Bummer that this isn't free for open source repositories, but I understand why. Looks like costs are pretty minimal, and free for the first (up-to) 60 hours per month.

This is definitely something we should still be vigilant about. It's easy to "leave the stove running" on this switching contexts to and from ReactUSWDS

Reviewer 3: Nothing to change on the PR, the step is explicit, just a thought I had while reading and learning about Codespaces (I haven't used them before!)
